src/sql/base/browser/ui/table/tableDataView.ts

```diff
@@ -79,20 +114,39 @@ export class TableDataView<T extends Slick.SlickData> implements IDisposableData
 		return this._data[index];
 	}
 
+	getLengthNonFiltered(): number {
+		return this.filterEnabled ? this._allData.length : this._data.length;
```

Comment:
this method name implies we would always want `allData`'s length

Reply:
to reduce memory usage, when filter is not turned on, we only use the _data property to store data. that's why this condition check.